### PR TITLE
Add `API::Pagination` concern

### PR DESCRIPTION
**Pull request by mjankowski**
Several related changes:

- Add an `API::Pagination` concern, and put the existing pagination methods from the api base controller into it
- Add an rspec matcher which wraps up the logic of comparing expected url values to the prev/next header link values, and replace previous specs with this approach -- in some cases this served to remove what was basically a double request call when the checks were in different examples
- Basically every API conrtoller that supports pagination had the same exact method definition of the `insert_pagination_headers` before_action. Move this to the newly created concern.

I think there is room for more future improvement in this area, but didn't want to go much further than this in first pass.

### CODE REVIEW

**Comment by mjankowski** [comment]
The two linked PRs here show some possible future additions to this concern.

**Comment by mjankowski** [comment]
Rebased this after the max/since one merged. This is probably ready to go as well, but let me know if this would be easier to review in smaller chunks. I could probably pull out the new matcher and spec changes into separate PR, and possibly do the concern changes in 2-3 PRs as well.

Separately, the linked pagination params PR - https://github.com/mastodon/mastodon/pull/28845 - should either be merged first and then incorporated into this change; or if this is merged first, it should be updated to move things to the concern instead of the base class.

**Comment by mjankowski** [comment]
Rebased after https://github.com/mastodon/mastodon/pull/29606 merged - this is now even smaller/cleaner, just moving the pagination-related methods already moved to base class, out to concern.

It would probably be best to review https://github.com/mastodon/mastodon/pull/28845 first, and then rebase this after that one.

**Review by ClearlyClaire — APPROVED** [review]
